src/init.cpp

```diff
@@ -1029,8 +1079,12 @@ bool AppInit2(boost::thread_group& threadGroup)
                 pcoinscatcher = new CCoinsViewErrorCatcher(pcoinsdbview);
                 pcoinsTip = new CCoinsViewCache(pcoinscatcher);
 
-                if (fReindex)
+                if (fReindex) {
                     pblocktree->WriteReindexing(true);
+                    //If we're reindexing in prune mode, wipe away all our block and undo data files
```

Comment:
I agree it doesn't effectively change any behaviour of -reindex, but I'd argue that -reindex itself has unexpected behaviour already in case of pruning. Anyway, not a big concern - maybe we can improve this after merge but before release.